### PR TITLE
[MAINTENANCE] RBP testing framework changes

### PR DESCRIPTION
**Pull request by Shinnnyshinshin**
# What's changed? 
- First of all, please don't be afraid of the size of diff. This PR is mostly the result of moving things around 😄 

1. Integration tests have been moved `tests/integration/profiling/rule_based_profilers/`, with associated fixtures living in `tests/integration/profiling/rule_based_profilers/conftest.py`. I defined integration tests as anything having to do with `run_profiler()` or the use of the configuration files `alice`, `bobby` and `bobster`, and associated fixtures were migrated over.  
2. The only exception are the tests in `tests/rule_based_profiler/test_user_workflow_fixtures.py`, which use the fixtures `alice` and `bobby`, but function more like unittests than integration tests. For these tests, the fixtures are loaded individually from the `conftest.py` file in the `tests/integration/profiling/rule_based_profilers/` folder. 
3. Moved fixtures in from the `tests/test_fixtures/rule_based_profiler/alpha` folder to the `tests/test_fixtures/rule_based_profiler/` folder. 
4. added `README.md` file to the `tests/test_fixtures/rule_based_profiler/` folder with descriptions of the `alice`, `bobby` and `bobster` fixtures. 

### Definition of Done
Please delete options that are not relevant.
- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/en/latest/contributing/style_guide.html?highlight=style%20guide)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html?highlight=checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

File: tests/test_fixtures/rule_based_profiler/README.md
Comment (Shinnnyshinshin) [inline tests/test_fixtures/rule_based_profiler/README.md:9]: I've added a short readme for the RBP configurations for Alice, Bobby and Bobster. Please feel to suggest additions or changes.

File: tests/rule_based_profiler/domain_builder/test_domain_builder.py
Comment (cdkini) [inline tests/rule_based_profiler/domain_builder/test_domain_builder.py]: `conftest` allows us to implicitly use fixtures without explicit importing. Could we get rid of this and just reference the fixtures in our test signatures?

Comment (cdkini) [reply tests/rule_based_profiler/domain_builder/test_domain_builder.py]: Ah this is necessary because the fixtures are for the integration tests? Could we add a small comment noting why we're doing importing from `conftest` just for some clarity?

**Review by cdkini — APPROVED** [review]
This looks great! Definitely appreciate the README ❤️ 

I'd probably let @NathanFarmer or @alexsherstinsky take a look before merging since they weighed in during your call but I think this is a nice improvement.

File: tests/integration/profiling/rule_based_profilers/conftest.py
Comment (cdkini) [inline tests/integration/profiling/rule_based_profilers/conftest.py]: Do you have any ideas on how to get rid of this pattern? We added this check to basically skip the test if the version was below 3.7 (due to dataclasses) but there has to be a better way to do this.

Instead of explicitly adding this to each test, we should be able to modify something at the `conftest` level. Not a blocker for this PR but I'd really love to see a more idiomatic pattern here.

File: tests/rule_based_profiler/domain_builder/test_domain_builder.py
Comment (NathanFarmer) [reply tests/rule_based_profiler/domain_builder/test_domain_builder.py]: I think I have convinced myself that it is correct to keep the integration tests in a separate directory like this (we need to test integration of various parts of the codebase). I'm just wondering if we should duplicate the fixtures or create new ones so we aren't importing between integration and unit tests. I think creating silos between test directories as much as possible is preferred due to the shared state issues we want to avoid (@cdkini please correct me if this understanding is wrong).

File: tests/rule_based_profiler/parameter_builder/test_simple_date_format_string_parameter_builder.py
Comment (NathanFarmer) [inline tests/rule_based_profiler/parameter_builder/test_simple_date_format_string_parameter_builder.py]: Wondering if we should duplicate these fixtures and move into unit test `conftest`.

File: tests/rule_based_profiler/test_user_workflow_fixtures.py
Comment (NathanFarmer) [inline tests/rule_based_profiler/test_user_workflow_fixtures.py]: Also here.

**Review by NathanFarmer — CHANGES_REQUESTED** [review]
Looks good based on what we have discussed so far, I just have one more concern about the fixture imports.

File: tests/rule_based_profiler/parameter_builder/test_simple_date_format_string_parameter_builder.py
Comment (Shinnnyshinshin) [reply tests/rule_based_profiler/parameter_builder/test_simple_date_format_string_parameter_builder.py]: @NathanFarmer I see your point, but I'm just a bit uncomfortable with duplicating fixtures. Another option may be to move this to the `conftest.py` in the base `tests/` directory

@anthonyburdi @cdkini @alexsherstinsky would would we be able to get your thoughts on this point?

Comment (NathanFarmer) [reply tests/rule_based_profiler/parameter_builder/test_simple_date_format_string_parameter_builder.py]: @Shinnnyshinshin After our internal discussions I think we have agreed to move these into `tests/test_fixtures` and that should avoid the imports.

File: tests/conftest.py
Comment (Shinnnyshinshin) [inline tests/conftest.py:5380]: Integration-test fixtures have been moved to `tests/conftest.py` eliminating the need for additional imports

File: tests/rule_based_profiler/domain_builder/test_domain_builder.py
Comment (NathanFarmer) [inline tests/rule_based_profiler/domain_builder/test_domain_builder.py:18]: ❤️

**Review by NathanFarmer — APPROVED** [review]
LGTM! Thanks for taking this thankless task on!

**Pull request merged**
